route.go

```diff
@@ -329,3 +335,101 @@ func (rm *RouteMetrics) encode(ae *netlink.AttributeEncoder) error {
 
 	return nil
 }
+
+// RTNextHop represents the netlink rtnexthop struct (not an attribute)
+type RTNextHop struct {
+	Length  uint16 // length of this hop including nested values
+	Flags   uint8  // flags defined in rtnetlink.h line 311
+	Hops    uint8
+	IfIndex uint32 // the interface index number
+}
+
+// NextHop wraps struct rtnexthop to provide access to nested attributes
+type NextHop struct {
+	Hop     RTNextHop // a rtnexthop struct
+	Gateway net.IP    // that struct's nested Gateway attribute
+	Value   []byte    // raw multipath `value` bytes (temporary for debugging)
+}
+
+// The RT_MULTIPATH netlink attribute contains a payload of `array of struct rtnexthop`
+type RTMultiPath []NextHop
+
+// consider instead creating an rtnetlink.MultiPathDecoder type
+// analogous to netlink.AttributeDecoder
+func (mp *RTMultiPath) decode(ad *netlink.AttributeDecoder) error {
+	const sizeOfRTNextHop = 8 // an rtNextHop is 8 bytes wide
+	// get RTA_Multipath data
+	mpData := ad.Bytes()
+	mpPayloadSize := len(mpData) // width of the multipath `value`(payload)
+
+	// check for truncated message
+	if mpPayloadSize <= sizeOfRTNextHop {
+		return errInvalidRouteMessageAttr
+	}
+
+	// Iterate through the nested array of rtnexthop, unpacking each and appending them to mp
+	for i := 0; i <= mpPayloadSize; {
+		// check for end of message
+		payloadRemaining := mpPayloadSize - i
+		if payloadRemaining < sizeOfRTNextHop {
+			return nil
+		}
+
+		// Copy over the struct portion
+		nh := NextHop{
+			Hop: RTNextHop{},
+		}
+		copy(
+			(*(*[sizeOfRTNextHop]byte)(unsafe.Pointer(&nh.Hop)))[:],
+			(*(*[sizeOfRTNextHop]byte)(unsafe.Pointer(&mpData)))[i:],
```

Comment:
Pretty sure this logic is incorrect. You'd want to point directly at the underlying array, but you're currently accessing the slice header itself.

I imagine it'd be something like `(*(*[sizeOfRTNextHop]byte)(unsafe.Pointer(&mpData[i:i+sizeOfRTNextHop])))[:]` instead.

This code is definitely going to need some fuzzing and careful tests.

Reply:
Is the sizeOfRTNextHop value constant on every platform or is it possible that it varies across kernels/architectures? If so, we should use x/sys/unix for that constant.

Reply:
the above syntax fails to build with `cannot take the address of mpData[i:i + sizeOfRTNextHop]`
I can get it building with an additional copy() by creating a next-hop-sized array to copy from eg.. 

```
var mpSlice [sizeOfRTNextHop]byte
        copy(mpSlice[:], mpData[i:i+sizeOfRTNextHop])

        copy(
            (*(*[sizeOfRTNextHop]byte)(unsafe.Pointer(&nh.Hop)))[:],
            (*(*[sizeOfRTNextHop]byte)(unsafe.Pointer(&mpSlice)))[:],
        )
```
is that sane? 

also, thanks, I'll change `sizeOfRTNextHop` to use the x/sys/unix constant.